sherlock.h

```diff
@@ -181,8 +218,28 @@ class StreamInstanceImpl {
           }
           bool user_initiated_terminate = false;
           data.ImmutableUse([&listener, &cursor, &user_initiated_terminate](const std::vector<T>& data) {
-            // TODO(dkorolev): RTTI dispatching here.
-            if (!listener->Entry(data[cursor])) {
+            // Entries are often instances of polymorphic types, that make it into various message queues.
+            // The most straightforward way to store them is a `unique_ptr`, and the most straightforward
+            // way to pass `unique_ptr`-s between threads is via `Emplace*(ptr.release())`.
+            // If `Entry()` is being passed immutable records, the pain of cloning data from `unique_ptr`-s
+            // becomes the user's pain. This shall not be allowed.
+            //
+            // Thus, here we need to make a copy.
+            // The below implementation is imperfect, but it serves the purpose semantically.
+            // TODO(dkorolev): Fix it.
+
+            const std::string json = JSON(data[cursor]);
+            T copy_of_entry;
+            try {
+              ParseJSON(json, copy_of_entry);
+            } catch (const std::exception& e) {
```

Comment:
The stream can be easily killed if it be being fed with data that can not be de-serialized right after being serialized.

On the backend, preventing data corruption has priority. Yes, it's a good design.